railties/lib/rails/generators/rails/app/interactive_options.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Rails
+  module Generators
+    class InteractiveOptions # :nodoc:
+      include Database
+      include Thor::Shell
+
+      attr_reader :options
+
+      def self.perform(initial_options)
+        new(initial_options).perform
+      end
+
+      def initialize(initial_options)
+        @options = initial_options
+      end
+
+      def perform
+        say "Rails CLI #{Rails.version}", :green
+
+        case inquire_rails_version
+        when "dev"    then options[:dev] = true
+        when "edge"   then options[:edge] = true
+        when "main" then options[:main] = true
+        end
+
+        options[:database]             = inquire_database
+        options[:skip_action_cable]    = inquire_action_cable
+        options[:skip_action_mailer]   = inquire_action_mailer
+        options[:skip_action_mailbox]  = inquire_action_mailbox
+        options[:skip_active_storage]  = inquire_active_storage
+        options[:skip_action_text]     = inquire_action_text
```

Comment:
Keep in mind that there are interdependencies between some components.  For example, Action Text and Active Storage.  So if a user says "no" to Active Storage, then we shouldn't ask them about Action Text.  Or we should ask them about Action Text first, then, if they say "yes", assume "yes" for Active Storage.

Reply:
I got here generally thinking about rails interactive in general. Like, Daniel Kehoe's rails-composer. Which asked you if you want haml vs erb, what db you want, minitest vs rspec, etc. 

It's a shame because it might be a big reach for rails core to approve a realistic interactive script that could install tooling that isn't a part of rails core. 

But my thinking here is this could be the start of an interactive script we build on.

However I can say it's definitely something I _would_ use. As a substitute to going to the rails docs and trying to review all the flags I have available to me. Component/modular/minimalist devs might feel the same?

Reply:
>  We could then open the file in the user's editor a la git rebase -i, with instructions to uncomment the options they wish to use

I think this would be a really cool idea